cli/command/stack/deploy_composefile.go

```diff
@@ -321,7 +321,16 @@ func deployServices(
 			if resolveImage == resolveImageAlways || (resolveImage == resolveImageChanged && image != service.Spec.Labels[convert.LabelImage]) {
 				updateOpts.QueryRegistry = true
 			}
-
+			if !updateOpts.QueryRegistry {
+				serviceSpec.TaskTemplate.ContainerSpec.Image = service.Spec.TaskTemplate.ContainerSpec.Image
+
+				if service.Spec.TaskTemplate.Placement != nil {
+					if serviceSpec.TaskTemplate.Placement == nil {
+						serviceSpec.TaskTemplate.Placement = &swarm.Placement{}
+					}
+					serviceSpec.TaskTemplate.Placement.Platforms = service.Spec.TaskTemplate.Placement.Platforms
```

Comment:
I considered moving this to the ServiceUpdate function but this problem only occurs with stack deploy, because it doesn't "diff" the same way as `docker service update` does. `docker service update` takes the existing service as a starting point, and only makes changes set through the `-add` / `-rm` flags, whereas stack deploy takes the compose-file as source of truth